src/tests/classes.test.js

```diff
@@ -1,3 +1,326 @@
+      slots: 9,
+    };
+
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510624').send(classData);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('If trainer or activity IDs are not found, return 404 and an error message', async () => {
+    const classData = {
+      day: 'Wednesday',
+      hour: '10:30',
+      trainer: '746642acfac4c6a035b35000',
+      activity: '7462261d7ead90b46b7471cc',
+      slots: 9,
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510624').send(classData);
+    expect(response.status).toBe(404);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Invalid ID should return status 400', async () => {
+    const response = await request(app).put('/api/classes/123456').send();
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('ID not found should return status 404', async () => {
+    const updatedClass = {
+      day: 'Monday', hour: '20:00',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510629').send(updatedClass);
+    expect(response.status).toBe(404);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Duplicated class should return status 400', async () => {
+    const duplicatedClass = {
+      day: 'Sunday', hour: '20:00',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(duplicatedClass);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Day not valid should return status 400', async () => {
+    const dayInvalid = {
+      day: 'Miernes',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(dayInvalid);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('Hour not valid should return status 400', async () => {
+    const hourInvalid = {
+      hour: 'asd023',
+    };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510620').send(hourInvalid);
+    expect(response.status).toBe(400);
+    expect(response.body.data).toBeUndefined();
+    expect(response.body.error).toBeTruthy();
+  });
+
+  test('If no changes were made, return status 400', async () => {
+    const notChanged = { day: 'Thursday' };
+    const response = await request(app).put('/api/classes/74663d50bb2d87b9f6510624').send(notChanged);
+    expect(response.status).toBe(200);
+    expect(response.body.data).toBeDefined();
+    expect(response.body.error).toBeFalsy();
+  });
```

Comment:
here it should return status 400 because status 200 is when the request was successful

Reply:
> here it should return status 400 because status 200 is when the request was successful

Wouldn't it be 200? 400 is when the request could not be processed and in this case the request does pass

Reply:
i think its as fede says, the error should be false so the status should be 200

Reply:
> > here it should return status 400 because status 200 is when the request was successful
> 
> Wouldn't it be 200? 400 is when the request could not be processed and in this case the request does pass

I also consider that 200 is more appropiate because the request itself doesn't have an error in that case, there were just no changes